src/LibraryViewExtension/LibraryViewController.cs

```diff
@@ -299,20 +300,35 @@ private void OnBrowserConsoleMessage(object sender, ConsoleMessageEventArgs e)
             System.Diagnostics.Trace.Write(e.Message);
         }
 
-        internal static IDisposable SetupSearchModelEventsObserver(NodeSearchModel model, IEventController controller, ILibraryViewCustomization customization, int throttleTime = 200)
+        internal static IDisposable SetupSearchModelEventsObserver(CustomNodeManager customNodeManager,NodeSearchModel model, IEventController controller, ILibraryViewCustomization customization, int throttleTime = 200)
         {
             customization.SpecificationUpdated += (o, e) => controller.RaiseEvent("libraryDataUpdated");
 
             var observer = new EventObserver<NodeSearchElement, IEnumerable<NodeSearchElement>>(
-                    elements => NotifySearchModelUpdate(customization, elements), CollectToList
+                    elements => NotifySearchModelUpdate(customization,elements), CollectToList
                 ).Throttle(TimeSpan.FromMilliseconds(throttleTime));
 
-            Action<NodeSearchElement> onRemove = e => observer.OnEvent(null);
+            Action<NodeSearchElement> handler = (searchElement) =>
+             {
+                // If the searchElement is customNodeSearchElement, only raise events
+                // to the library if it has been registered - when this has occured the customNodeManager will
+                // have its workspace loaded - if not, return.
+                if (searchElement is CustomNodeSearchElement)
+                 {
+                     if (customNodeManager.GetWorkspaceById((searchElement as CustomNodeSearchElement).ID) == null)
+                     {
+                         return;
+                     }
+                 }
+
+                 observer.OnEvent(searchElement);
+             };
+            Action<NodeSearchElement> onRemove = e => handler(null);
 
             //Set up the event callback
-            model.EntryAdded += observer.OnEvent;
+            model.EntryAdded += handler;
             model.EntryRemoved += onRemove;
-            model.EntryUpdated += observer.OnEvent;
+            model.EntryUpdated += handler;
```

Comment:
the disposed should say model.EntryUpdated -= handler?

Reply:
I will look into removing these handlers when the library is disposed.